ytfzf

```diff
@@ -571,7 +571,31 @@ set_vars () {
     #this comes from invidious' api
     thumbnail_quality="high"
     sub_link_count="2"
-    invidious_instance="https://y.com.sb" yt_video_link_domain="https://youtube.com"
+
+    # URL for instance list sorted by type and health
+    instances_url="https://api.invidious.io/instances.json?sort_by=type,health" 
+
+    # Cache location.
+    # TODO: decide on cache interval to avoid instances.json becoming stale
+    instances_file="$cache_dir/instances.json"
+
+    # If file doesn't already exist, cURL and cache it.
+    [ ! -f "$instances_file" ] && curl -X GET -s "$instances_url" > "$instances_file"
+ 
+    # The pipeline does the following:
+    #   - Get top 10 instance URIs
+    #   - Trim out extra symbols produced by `jq`
+    #   - Remove empty lines
+    #   - Shuffle the list
+    #   - Choose the first entry
+    invidious_instance="$(jq -r 'map(.[1] | .uri) | .[0:10]' < "$instances_file" \
+                        | tr -d '[]," ' \
+                        | sed "/^$/d" \
+                        | shuf \
+                        | head -n1)"
```

Comment:
I believe this can be rewritten as:

```sh
invidious_instance=$(jq -r 'map(.[1] | .uri) | .[:10] | join("\n") < "$instances_file" \
                      | shuf | head -n 1)
```

Reply:
Yup, you're right. Code now reflects this.